### PR TITLE
 Added Matrix folder in C

### PR DESCRIPTION
**Pull request by ayanujju**
Fixed#450; added matrix addition,subtraction,multiplication,sorting,scalar product, etc.

### CODE REVIEW

**Comment by ayanujju** [comment]
Fixed

**Comment by ayanujju** [comment]
Please tell me, is there  any other modification required? If not, please merge my PR.

File: Matrix_Operations/Matrix Operation in C [hunk 1]
```diff
@@ -0,0 +1,515 @@
+/* C Program to Add Two Matrices using Multi-dimensional Arrays */
+
+
+#include <stdio.h>
+
+int main()
+{
+    int r, c, a[100][100], b[100][100], sum[100][100], i, j;
```
Comment (jainaman224) [inline Matrix_Operations/Matrix Operation in C]: Please use descriptive variable names.

**Comment by ayanujju** [comment]
Please help me, I have divided all the different operations in different files but mistakenly created pull request

File: Matrix_Operations/C/Matrix_sort.c [hunk 2]
```diff
@@ -0,0 +1,148 @@
+/* C Program to sort a matrix */
+
+
+
+#include <stdio.h>
+ 
+int main () {
+ 
+	static int ma[10][10],mb[10][10];
```
Comment (jainaman224) [inline Matrix_Operations/C/Matrix_sort.c]: Use 4 space indentation.
Change file name to Matrix_Sort.c

File: Matrix_Operations/C/Matrix_determinant.c [hunk 3]
```diff
@@ -0,0 +1,98 @@
+/* C Program to compute determinant of a Matrix */
```
Comment (jainaman224) [inline Matrix_Operations/C/Matrix_determinant.c]: Change file name to Matrix Determinant.

**Comment by ayanujju** [comment]
Please review it. I have made suggested changing.

File: Matrix_Operations/C/Matrix_Sort.c [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+ 
+	printf ("Enter the order of the matrix \n");
+ 
+	scanf ("%d %d", &m,&n);
+ 
+	printf ("Enter co-efficients of the matrix \n");
+ 
+	for (i=0;i<m;++i) {
+ 
+		for (j=0;j<n;++j) {
```
Comment (jainaman224) [inline Matrix_Operations/C/Matrix_Sort.c]: 8 space indentation.

File: Matrix_Operations/C/Matrix Determinant [hunk 5]
```diff
@@ -0,0 +1,98 @@
+/* C Program to compute determinant of a Matrix */
```
Comment (jainaman224) [inline Matrix_Operations/C/Matrix Determinant]: file name should be Matrix_Determinant.c

**Comment by ayanujju** [comment]
Please check it.

File: Matrix_Operations/C/Matrix_Addition.c [hunk 6]
```diff
@@ -0,0 +1,81 @@
+/* C Program to Add Two Matrices using Multi-dimensional Arrays */
+
+
```
Comment (marwahmanbir) [inline Matrix_Operations/C/Matrix_Addition.c]: Remove unnecessary blank lines like these.

File: Matrix_Operations/C/Matrix_Addition.c [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+Enter elements of 2nd matrix:
+Enter element b11: 3
+Enter element b12: -9
+Enter element b21: 4
+Enter element b22: -9
+Sum of two matrix is:
+10   -4
+7   -3
+*/
+
```
Comment (marwahmanbir) [inline Matrix_Operations/C/Matrix_Addition.c]: Keep only single blank line before & after output.

File: Matrix_Operations/C/Matrix_Sort.c [hunk 8]
```diff
@@ -0,0 +1,146 @@
+/* C Program to sort a matrix */
+
+
+
+#include <stdio.h>
+ 
+int main () 
+{
+    
+	static int ma[10][10],mb[10][10];
```
Comment (marwahmanbir) [inline Matrix_Operations/C/Matrix_Sort.c]: Fix indentation - 4 spaces instead of tab.

File: Matrix_Operations/C/Matrix_Sort.c (hunk 4 above)
Comment (marwahmanbir) [reply Matrix_Operations/C/Matrix_Sort.c]: This is also not resolved yet.

File: Matrix_Operations/C/Matrix_Sort.c [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+ 
+	scanf ("%d %d", &m,&n);
+ 
+	printf ("Enter co-efficients of the matrix \n");
+ 
+	for (i=0;i<m;++i) {
+ 
+		for (j=0;j<n;++j) {
+ 
+			scanf ("%d",&ma[i][j]);
```
Comment (marwahmanbir) [inline Matrix_Operations/C/Matrix_Sort.c]: 12-space indentation

File: Matrix_Operations/C/Matrix_Sort.c [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+ 
+	printf ("After arranging rows in ascending order\n");
+ 
+	for (i=0;i<m;++i) {
+ 
+		for (j=0;j<n;++j) {
+ 
+			for (k=(j+1);k<n;++k) {
+ 
+				if (ma[i][j] > ma[i][k]) {
```
Comment (marwahmanbir) [inline Matrix_Operations/C/Matrix_Sort.c]: 16-space indentation.

File: Matrix_Operations/C/Matrix_Sort.c [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+ 
+					a = ma[i][j];
```
Comment (marwahmanbir) [inline Matrix_Operations/C/Matrix_Sort.c]: 20-space indentation.

File: Matrix_Operations/C/Matrix_Sort.c [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+			}
+ 
+		}
+ 
+	}
+ 
+	/* End of outer for loop*/
+ 
+	for (i=0;i<m;++i) {
+
```
Comment (marwahmanbir) [inline Matrix_Operations/C/Matrix_Sort.c]: Don't use blank line after each statement. Add space only at
- One in-between each block
- One before every comment
- Before and after output
- After header files/define statement

File: Matrix_Operations/C/Matrix_Subtraction.c [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+	   scanf("%d",&a[i][j]);
+       }
+    // Storing elements of second matrix entered by user
+    
+    printf("\n "Enter elements of 2nd matrix: \n");
+    for(i = 0; i < r; ++i)
+       for(j = 0; j < c; ++j)
+       {
+           printf("Enter element b%d%d:",i+1,j+1);
+	   scanf("%d",&b[i][j]);
```
Comment (marwahmanbir) [inline Matrix_Operations/C/Matrix_Subtraction.c]: Fix indentation.

File: Matrix_Operations/C/Matrix_Subtraction.c [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+            diff[i][j] = a[i][j] - b[i][j];
+
+    // Displaying the resultant sum matrix
+    
+    printf("\nDifference of two matrix is: \n");
+    
+    for(i = 0; i < r; ++i)
+        for(j = 0; j < c; ++j)
+        {
+	    printf("%d ",sum[i][j]);
```
Comment (marwahmanbir) [inline Matrix_Operations/C/Matrix_Subtraction.c]: Fix indentation.

**Comment by marwahmanbir** [comment]
Review Summary:
1. Remove unnecessary blank lines.
2. Fix indentation, 4 spaces in place of 1 tab, 8 spaces in place of 2 tabs and so on. Most IDEs/text editors have an option to replace tabs with spaces. Do a Google search about how to fix using the text editor you're using.
3. Don't use indentation where not necessary.

You can check out recently merged PRs to get help for fixing your files.

File: Matrix_Operations/C/Matrix_Sort.c (hunk 12 above)
Comment (marwahmanbir) [reply Matrix_Operations/C/Matrix_Sort.c]: ... and so on (wherever necessary)

**Comment by jainaman224** [comment]
@ayanujju If you are facing issues. You can connect with us on slack.

**Pull request closed without merging**